### PR TITLE
Fix exception when searching

### PR DESCRIPTION
**Pull request by Fifi-wang**
We fix the issue that uncaught exception when searching in dark mode. In fact, we found that this issue is not a dark mode issue and this is a threading issue. So we use Platform.runLater() to ensure that there will be no conflicts when adding key-value pairs to the map.
fixes [#7343](https://github.com/JabRef/jabref/issues/7343)

- [x] Change in `CHANGELOG.md` described in a way that is understandable for the average user (if applicable)
- [ ] Tests created for changes (if applicable)
- [x] Manually tested changed features in running JabRef (always required)
- [ ] Screenshots added in PR description (for UI changes)
- [x] [Checked documentation](https://docs.jabref.org/): Is the information available and up to date? If not created an issue at <https://github.com/JabRef/user-documentation/issues> or, even better, submitted a pull request to the documentation repository.

### CODE REVIEW

**Comment by tobiasdiez** [comment]
Thanks a lot for the investigation and the fix! 

However, I fear that the proposed changes will lead to other threading problems because currently `setField` is used with the assumption that changes are immediately stored in the entry. This is no longer the case when using `Platform.runLater`. Thus, I would propose to fix this at a later stage in the call hierarchy, e.g. in the import dialog. Do you know which code exactly leads to the threading issue?

**Comment by Fifi-wang** [comment]
> Thanks a lot for the investigation and the fix!
> 
> However, I fear that the proposed changes will lead to other threading problems because currently `setField` is used with the assumption that changes are immediately stored in the entry. This is no longer the case when using `Platform.runLater`. Thus, I would propose to fix this at a later stage in the call hierarchy, e.g. in the import dialog. Do you know which code exactly leads to the threading issue?

I apologize for not considering other thread issues. The threading problem here mainly occurs in the put and remove methods I modified. Because the ObservableMap used here is not a thread-safe mapping. Therefore, there will be conflicts between put and remove.

**Comment by tobiasdiez** [comment]
No worries!

Do you know which code is invoking put and remove from different threads?

**Comment by Fifi-wang** [comment]
org.jabref.logic.importer.ImportCleanup#doPostCleanup(java.util.Collection<org.jabref.model.entry.BibEntry>)
This function use ParallelStream to invoke doPostCleanup function where make thread conflicts.
org.jabref.model.entry.BibEntry#clearField(org.jabref.model.entry.field.Field, org.jabref.model.entry.event.EntriesEventSource)
org.jabref.model.entry.BibEntry#setField(org.jabref.model.entry.field.Field, java.lang.String, org.jabref.model.entry.event.EntriesEventSource)
These two functions will invoke put and remove directly.

**Comment by tobiasdiez** [comment]
If the parallel stream is the source for the problems, then just use a normal stream.

**Review by tobiasdiez — APPROVED** [review]
Thanks for the quick follow-up!

**Review by Siedlerchr — APPROVED** [review]

File: CHANGELOG.md [hunk 1]
```diff
@@ -524,6 +524,7 @@ Note that this project **does not** adhere to [Semantic Versioning](http://semve
 - We fixed an issue where the "Search for unlinked local files" would throw an exception when parsing the content of a PDF-file with missing "series" information [#5128](https://github.com/JabRef/jabref/issues/5128)
 - We fixed an issue where the XMP Importer would incorrectly return an empty default entry when importing pdfs [#6577](https://github.com/JabRef/jabref/issues/6577)
 - We fixed an issue where opening the menu 'Library properties' marked the library as modified [#6451](https://github.com/JabRef/jabref/issues/6451)
+- We fixed an issue where has exception when importing library [#7343](https://github.com/JabRef/jabref/issues/7343)
```
Comment (Siedlerchr) [inline CHANGELOG.md:527]:
```suggestion
- We fixed an issue when importing resulted in an exception [#7343](https://github.com/JabRef/jabref/issues/7343)
```

**Pull request merged**